### PR TITLE
HPCC-14433 Allow user <hpcc> to run hpcc-init and dafilesrv

### PR DESCRIPTION
**Pull request by xwang2713**
<hpcc> user cannot run hpcc-init and dafilesrv since these script only allow root user to run.
But we used to allow <hpcc> user to start these services which is also helpful in non-root user privilege case.  
To fix it:
1) remove is_root in hpcc-init and dafilesrv
2) move creating /var/log/HPCCSystems to install-init script
3) remove "su <hpcc> -c " if current user is <hpcc> user to prevent prompts for password
4) Make sure debug log file owned by <hpcc> user

Signed-off-by: Xiaoming Wang xiaoming.wang@lexisnexi.com

@Michael-Gardner  please review

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@xwang2713 I can't get this to work on a fresh installation.  The HPCCSystems directories in /var/lib, /var/run, /var/lock, /var/log need to be laid down w/ a chown hpcc:hpcc beforehand.

**Comment by xwang2713** [comment]
Add "createRuntime" to create create runtime directories and chown to <hpcc> user

File: initfiles/bash/etc/init.d/install-init.in [hunk 1]
```diff
@@ -300,3 +300,13 @@ fi
...
     date > ${CONFIG_DIR}/installed
 fi
+
+if [ ! -d ${LOG_DIR} ]; then
+   mkdir -p ${LOG_DIR}
+   chown ${user}:${user} ${LOG_DIR}
+fi
+[ -e  ${LOG_DIR}/hpcc-init.debug ] && chown ${user}:${user} ${LOG_DIR}/hpcc-init.debug
+
+createRuntime
```
Comment (Michael-Gardner) [inline initfiles/bash/etc/init.d/install-init.in]: Ming I think if you move this up to line 304, you can them remove the LOG_DIR block.  Im pretty sure it's redundant in the createRuntime function.  Other than that, this looks solid.  Tested it and it works fine.

Comment (xwang2713) [inline initfiles/bash/etc/init.d/install-init.in]: Actually "$compName", referenced by createRuntime, is not defined so install may throw error unless modifying createRuntime function. As user may change HPCC cluster topology I wonder if it is worthy we process the runtime setting during install.

**Comment by Michael-Gardner** [comment]
yeah the empty $compName could be an issue, but since we have it unset it should evaluate out to "".  I've checked createRuntime and it doesn't look like there should be an issue with an empty compName.  Not 100% certain though.  It did work in my tests.

**Comment by xwang2713** [comment]
@Michael-Gardner please help to review again for the change I made.

**Comment by richardkchapman** [comment]
This has missed the boat for 5.4.4 - will need to be rebased for 5.4.6 or 6.0.0 as appropriate

**Comment by xwang2713** [comment]
OK. I will rebase for 6.0.0

**Comment by xwang2713** [comment]
Will re-open against HPCC branch candidate-6.0.0

**Comment by xwang2713** [comment]
Will re-open for HPCC candidate-6.0.0

**Pull request closed without merging**